packages/runner/lib/monitor.ts

```diff
@@ -114,6 +114,20 @@ export class RunnerMonitor {
             this.checkIdle(nextTimeout);
         }, timeoutMs);
     }
+
+    hasConflictingSync(newTask: NangoProps): boolean {
+        if (newTask.scriptType !== 'sync') {
+            return false;
+        }
+
+        for (const task of this.tracked.values()) {
```

Comment:
I would have modified `tracked` to use syncId as the map key and fail `track()` if key already exists

Reply:
True, maybe it's a bit more explicit like this imo but I can change